src/scripts/client/modules/gradebook.js

```diff
@@ -1,424 +1,387 @@
-            } else {
-                studentRow.append( '<td></td>' );
-            }
-        }
-
-        return studentRow;
-    }
-
-    // Data Management +++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-
-    let idFromUrl = uri => Number( uri.substr( uri.lastIndexOf( '/' ) + 1 ) );
-
-    function calculateGrades( students, assignments ) {
-        if ( !students ) {
-            return;
-        }
-
-        // Question, What is the the best practice for building an uniq array
-        // of objects in JS.
-        Object.keys( students ).map( ( studentName ) => {
-            var student = students[ studentName ];
+    );
 
-            var submissions = Object.keys( student.submissions );
+    studentRow.append(
+      `<td class='grade-percent' title='Click here to toggle other student grades visibility'>${student.percentage}%</td>`
+    );
 
-            var okCount = submissions.filter( assignmentId => {
-                return OK_GRADES.includes( student.submissions[ assignmentId ].grade );
-            } );
+    assignments.forEach(function addStudentAssignmentGradeElement(assignment) {
+      const submission = student.submissions[assignment.id];
 
-            var grade = okCount.length / assignments.length * 100;
-            student.percentage = grade.toFixed( 0 );
-        } );
-        return students;
+      if (submission) {
+        const gradeClass = SHORT_GRADE_NAMES[submission.status].toLowerCase();
+        studentRow.append(
+          `
+          <td class='grade ${gradeClass}' date-tooltip='${assignment.title}' >
+            <a href='${submission.href}' title='${assignment.title}' target='blank' >
+              ${SHORT_GRADE_NAMES[submission.status]}
+            </a>
+          </td>
+          `
+        );
+      } else {
+        studentRow.append('<td></td>');
+      }
+    });
+
+    return studentRow;
+  }
+  // Data Management +++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  function calculateGrades(students, assignments) {
+    if (!students) {
+      return false;
     }
 
-    function getPathContent(path) {
-      const pathURI = id => `https://newline.theironyard.com/admin/paths/${ id }`;
+    // Question, What is the the best practice for building an uniq array
+    // of objects in JS.
+    return Object.keys(students).map(function calculateGradeForStudent(studentName) {
+      const student = students[studentName];
+
+      const submissions = Object.keys(student.submissions);
+      const okCount = submissions.filter(id => OK_GRADES.includes(student.submissions[id].status));
+      const grade = (okCount.length / assignments.length) * 100;
+      student.percentage = grade.toFixed(0);
+      return student;
+    });
+  }
+
+  function getPathContent(path) {
+    const pathURI = id => `https://newline.theironyard.com/api/paths/${id}/contents`;
+    return new Promise(function pathDataHandler(res, rej) {
+      $.get(pathURI(path.id)).then(function processPathData(data) {
```

Comment:
We're using `then` and `fail` here, but `done` in an ajax call earlier... which one should we use? I think they're synonyms?